file_handler.c

```diff
@@ -22,23 +26,29 @@ int is_guiding_line(char *line_pointer) {
 
     char *token = strtok(line, " ");
     if (token == NULL) {
-        return 0;
+        return 0; /* ##CR: Wasn't this case covered when we check if the line is empty?
+        I mean, token will be NULL at this point only if the line is empty. But if it's empty,
+        we'll not get to this part of the code anyway */
     }
     if (strcmp(token, ".data") == 0
         || strcmp(token, ".string") == 0
         || strcmp(token, ".struct") == 0
-        || strcmp(token, ".extern") == 0) {
-        return 1;
+        || strcmp(token, ".extern") == 0
+        || strcmp(token, ".entry") == 0) {
+        return 1; /* ##CR: It could also be a label. Need to add something like - if token is label --> check next word (labels have syntax rules that can be verified)
+        Also, we need a 1st round that only looks for symbols and stores them in a symbols table (it's emphasized in the instructions) */
     }
     token = strtok(NULL, " ");
     if (token == NULL) {
-        return 0;
+        return 0; /* ##CR: printf("missing arguments") 
+                     Also not that they emphasize that the program should keep running even when errors are encountered */
     }
     if (strcmp(token, ".data") == 0
         || strcmp(token, ".string") == 0
         || strcmp(token, ".struct") == 0
-        || strcmp(token, ".extern") == 0) {
-        return 1;
+        || strcmp(token, ".extern") == 0
+        || strcmp(token, ".entry") == 0) {
+        return 1; /* ##CR: duplicate */

```

Comment:
first is in case its without symbol, now its with symbols
I dont care much about best practices here because its a small uni project and it doesnt matter